### ISSUE TITLE
Regression in zabbix_agent/tasks/userparameter.yml when ansible.windows collection is not installed

### ISSUE THREAD

**Issue by weehal** [ansible-collections/community.zabbix#514]
##### SUMMARY
`zabbix-agent/task/userparameter.yml` fails when using ansible < 2.10  ansible.windows collection is not installed with error 

```json
FAILED! => {
  "reason": "couldn't resolve module/action 'ansible.windows.win_template'. This often indicates a misspelling, missing collection, or incorrect module path.
  
  The error appears to be in '/builds/(...)/collections/ansible_collections/community/zabbix/roles/zabbix_agent/tasks/userparameter.yml': line 3, column 5, but may
  be elsewhere in the file depending on the exact syntax problem.
  
  The offending line appears to be:
  - block:
    - name: \"Windows | Installing user-defined userparameters\"
      ^ here
  "
}

```
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`zabbix-agent/task/userparameter.yml` 

##### ANSIBLE VERSION

```console
ansible 2.9.21
  config file = (...)/projectX/ansible.cfg
  configured module search path = ['$HOME/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = $HOME/vEnvs/ansible/lib/python3.8/site-packages/ansible
  executable location = $HOME/vEnvs/ansible/bin/ansible
  python version = 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0]
```

##### CONFIGURATION

```console
ANSIBLE_COW_SELECTION($HOME/projectX/ansible.cfg) = random
ANSIBLE_NOCOWS($HOME/projectX/ansible.cfg) = True
ANSIBLE_PIPELINING($HOME/projectX/ansible.cfg) = True
ANSIBLE_SSH_ARGS($HOME/projectX/ansible.cfg) = -C -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s
CACHE_PLUGIN($HOME/projectX/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(env: ANSIBLE_CACHE_PLUGIN_CONNECTION) = ./.cache
CACHE_PLUGIN_TIMEOUT($HOME/projectX/ansible.cfg) = 14400
COLLECTIONS_PATHS(env: ANSIBLE_COLLECTIONS_PATHS) = ['$HOME/projectX/.galaxy/collections']
DEFAULT_CALLBACK_WHITELIST(env: ANSIBLE_CALLBACK_WHITELIST) = ['timer', 'profile_roles', 'profile_tasks']
DEFAULT_GATHERING($HOME/projectX/ansible.cfg) = smart
DEFAULT_HOST_LIST($HOME/projectX/ansible.cfg) = ['$HOME/projectX/inventory/hosts.ini']
DEFAULT_REMOTE_USER(env: ANSIBLE_REMOTE_USER) = warf
DEFAULT_ROLES_PATH(env: ANSIBLE_ROLES_PATH) = ['$HOME/projectX/roles', '$HOME/projectX/.galaxy/roles']
DEFAULT_STDOUT_CALLBACK(env: ANSIBLE_STDOUT_CALLBACK) = yaml
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = $HOME/.ansible/secrets/obc.vault
DIFF_ALWAYS(env: ANSIBLE_DIFF_ALWAYS) = True
DISPLAY_SKIPPED_HOSTS(env: ANSIBLE_DISPLAY_SKIPPED_HOSTS) = True
HOST_KEY_CHECKING($HOME/projectX/ansible.cfg) = False
INTERPRETER_PYTHON($HOME/projectX/ansible.cfg) = auto_legacy_silent
INVENTORY_ENABLED(env: ANSIBLE_INVENTORY_ENABLED) = ['yaml', 'ini']
PERSISTENT_COMMAND_TIMEOUT(env: ANSIBLE_PERSISTENT_COMMAND_TIMEOUT) = 30
PERSISTENT_CONNECT_TIMEOUT(env: ANSIBLE_PERSISTENT_CONNECT_TIMEOUT) = 20
RETRY_FILES_ENABLED(env: ANSIBLE_RETRY_FILES_ENABLED) = False
```

##### OS / ENVIRONMENT / Zabbix Version

Debian buster/ Ubuntu focal/  Zabbix 5.2 / 5.4

##### STEPS TO REPRODUCE

```yaml
- name: zabbix agent deployment
  hosts: db_mysql_nodes
  become: true
  strategy: free
  collections:
    - community.zabbix
  roles:
    - role: community.zabbix.zabbix_agent
       vars:
        zabbix_agent_userparameters:
          - name: mysql
      tags: zabbix-agent
```

##### EXPECTED RESULTS

In community.zabbix:1.4.0

```console
TASK [community.zabbix.zabbix_agent : Installing user-defined userparameters] ***********************************************************************************************************************************task path: $HOME/projektX/.galaxy/collections/ansible_collections/community/zabbix/roles/zabbix_agent/tasks/userparameter.yml:24
--- before
+++ after: $HOME/.ansible/tmp/ansible-local-5265g4t1cbpz/tmpg6trbipc/mysql.j2
@@ -0,0 +1,15 @@
+#template_db_mysql.conf created by Zabbix for "Template DB MySQL" and Zabbix 4.2
+#For OS Linux: You need create .my.cnf in zabbix-agent home directory (/var/lib/zabbix by default)
+#For OS Windows: You need add PATH to mysql and mysqladmin and create my.cnf in %WINDIR%\my.cnf,C:\my.cnf,BASEDIR\my.cnf https://dev.mysql.com/doc/refman/5.7/en/option-files.html
+#The file must have three strings:
+#[client]
+#user=zbx_monitor
+#password=<password>
+#
+UserParameter=mysql.ping[*], mysqladmin -h"$1" -P"$2" ping
+UserParameter=mysql.get_status_variables[*], mysql -h"$1" -P"$2" -sNX -e "show global status"
+UserParameter=mysql.version[*], mysqladmin -s -h"$1" -P"$2" version
+UserParameter=mysql.db.discovery[*], mysql -h"$1" -P"$2" -sN -e "show databases"
+UserParameter=mysql.dbsize[*], mysql -h"$1" -P"$2" -sN -e "SELECT SUM(DATA_LENGTH + INDEX_LENGTH) FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA='$3'"
+UserParameter=mysql.replication.discovery[*], mysql -h"$1" -P"$2" -sNX -e "show slave status"
+UserParameter=mysql.slave_status[*], mysql -h"$1" -P"$2" -sNX -e "show slave status"
```

##### ACTUAL RESULTS

```console
ansible-playbook playbooks/zbx-test.yml --limit dbhosts --tags zabbix,zabbix-agent,zabbix-agent-helper --check -vvvv
TASK [community.zabbix.zabbix_agent : Including userparameters] *************************************************************************************************************************************************
task path: $HOME/projektX/.galaxy/collections/ansible_collections/community/zabbix/roles/zabbix_agent/tasks/main.yml:106
fatal: [dbhost1]: FAILED! =>
  reason: |-
    couldn't resolve module/action 'ansible.windows.win_template'. This often indicates a misspelling, missing collection, or incorrect module path.

    The error appears to be in '$HOME/projektX/.galaxy/collections/ansible_collections/community/zabbix/roles/zabbix_agent/tasks/userparameter.yml': line 3, column 5, but may
    be elsewhere in the file depending on the exact syntax problem.

    The offending line appears to be:

    - block:
      - name: "Windows | Installing user-defined userparameters"
        ^ here
```

**Comment by dj-wasabi** [comment]
Hi @weehal 

A PR #477 is merged a while ago, that resulted in that Ansible 2.10 is the minimum version to be used with the roles in this collection. So I would suggest to either use 1.4.0 as version (if you can not use a newer version of Ansible), or upgrade to a new version of Ansible.

**Comment by weehal** [comment]
I managed to work around the problem by installing the `community.windows` and `ansible.windows` collections even though I am not using MS Windows in that environment.

BTW – when testing on Windows in other place I've noticed in `roles/zabbix_agent/tasks/Windows.yml`:

```yaml
- name: "Windows | Register Service"
  ansible.windows.win_command: '"{{ zabbix_win_exe_path }}" --config "{{ zabbix_win_install_dir }}\{{ zabbix_win_config_name }}" --install'
  when: falset zabbix_windows_service.exists
``` 

OFC it's a story for another issue.

**Comment by dj-wasabi** [comment]
Hi @weehal 

A PR is merged that fixes the typo. Can you confirm that it is now correctly?

**Comment by weehal** [comment]
the typo was not the error I was reporting, but I confirm that the typo is fixed

**Comment by dj-wasabi** [comment]
1.5.1 is created and available. I will close this issue.
